src/@context/Asset.tsx

```diff
@@ -58,110 +59,125 @@ function AssetProvider({
 
   const newCancelToken = useCancelToken()
 
-  const fetchDdo = async (token?: CancelToken) => {
-    LoggerInstance.log('[asset] Init asset, get DDO')
-    setLoading(true)
-    const ddo = await retrieveDDO(did, token)
-
-    if (!ddo) {
-      setError(
-        `[asset] The DDO for ${did} was not found in MetadataCache. If you just published a new data set, wait some seconds and refresh this page.`
-      )
-    } else {
-      setError(undefined)
-    }
-    setLoading(false)
-    return ddo
-  }
+  const fetchAsset = useCallback(
+    async (token?: CancelToken) => {
+      LoggerInstance.log('[asset] Init asset, get assetExtended')
+      setLoading(true)
+      const assetExtended = await retrieveDDO(did, token)
+
+      if (!assetExtended) {
+        setError(
+          `[asset] The assetExtended for ${did} was not found in MetadataCache. If you just published a new data set, wait some seconds and refresh this page.`
+        )
+      } else {
+        setError(undefined)
+      }
+      setLoading(false)
+      return assetExtended
+    },
+    [did]
+  )
 
-  const refreshDdo = async (token?: CancelToken) => {
+  const refreshAsset = async (token?: CancelToken) => {
```

Comment:
do we need this method at all? Seems everything it does could be done above in `fetchAsset`, then we just call this

Reply:
no idea, was used in compute, didn't have time to check it out